Dockerfile

```diff
@@ -0,0 +1,30 @@
+# vim:set ft=dockerfile:
+FROM python:3.6.3-slim
```

Comment:
I also switched from using `python:3.6` to `python:3.6.3-slim` as the base. We can update the python base image version as new versions of python are released. The `-slim` image is barebones and avoids a lot of bloat that we don't need. It is 156MB vs 690MB for the standard image. We are already making sure that we're installing all the OS dependencies we need. The resulting image is ~350MB vs ~750MB for the current synapse image.

From the docs at https://hub.docker.com/_/python/:
```
python:slim
This image does not contain the common packages contained in the default tag and only contains the minimal packages needed to run python. ...
```

Reply:
Does this include the psql drivers needed to run psycopg2?  We may want to include psycopg2 and paramiko in the synapse base image since they are delayed imports which would not be present in a install built from setup.py

Reply:
After discussion, I'll add both. Postgres is done,  paramiko in progress

Reply:
The SSH link has no unit tests and is (as far as I know) unused. I'm going to skip adding paramiko in this PR, as it already will not work in the existing images and there are no tests. I'll create an issue to track this, we either need to have coverage on that module or remove it.

Reply:
See https://github.com/vertexproject/synapse/issues/493